### ISSUE TITLE
Assistant is out of sync with page

### ISSUE THREAD

**Issue by marknelson** [draber/draber.github.io#109]
Mac, Chrome 98.0.4758.102, Spelling Bee Assistant 4.2.3

Assistant is only seeing words entered in the currently open session. Refreshing the page or entering words elsewhere results in a loss of the correct word count. Seems to be a new problem.

![spelling bee](https://user-images.githubusercontent.com/1194106/154525148-bfeff53c-1cfc-40d1-865e-8d00604a7cad.png)

**Comment by draber** [comment]
The fact that SBA is sometimes not syncing correctly between devices is a known problem that to my knowledge cannot be properly solved. Spelling Bee asynchronously pulls the game status from the server after the app has been launched. SBA cannot know reliably when this is finished so it relies on guess work.

What strikes me odd is that this persists even after a refresh. Is this something you see on a regular basis?

I'm currently working on a update in other areas of SBA and will look into the issue again, but I don't have very hopes, to be honest.

**Comment by marknelson** [comment]
>Is this something you see on a regular basis?

No, it's new, and right now I am seeing it consistently on OS X, not on Windows. But there's no fixing it on the Mac - refresh, reload, restart browser, doesn't matter - all my counts are stuck at zero.

**Comment by draber** [comment]
OK, thanks for the info, I'll look into it

**Comment by draber** [comment]
Unfortunately I cannot replicate the problem.

I tested the following scenario with SBA 4.2.3
- Played on my phone (Kiwi browser/Android) until I had 5 words left
- Launched game on Chrome 98/macOS 12.1, no extensions other than SBA, **synced OK**
- Launched game on Chrome 98 (Brave) on Windows 10/64, **synced OK**

Do you get any error messages or anything else I could work with?

**Comment by marknelson** [comment]
Yeah actually I might. I have assumed these errors are not in the extension but in the spelling-bee page, but possibly they are related. There is this:
foundation.4cb046f7c95f15a4188f27bdda945fb9312730b1.js:1 local storage failure: DOMException: Failed to execute 'setItem' on 'Storage': Setting the value of 'sb-today' exceeded the quota.
    at E (https://www.nytimes.com/games-assets/v2/spelling-bee.4cb046f7c95f15a4188f27bdda945fb9312730b1.js:1:2505)
    at o (https://www.nytimes.com/games-assets/v2/spelling-bee.4cb046f7c95f15a4188f27bdda945fb9312730b1.js:1:984)
    at https://www.nytimes.com/games-assets/v2/spelling-bee.4cb046f7c95f15a4188f27bdda945fb9312730b1.js:1:1511
    at v (https://www.nytimes.com/games-assets/v2/react-bundle.4cb046f7c95f15a4188f27bdda945fb9312730b1.js:276:3865)
    at https://www.nytimes.com/games-assets/v2/spelling-bee.4cb046f7c95f15a4188f27bdda945fb9312730b1.js:1:41189
    at https://www.nytimes.com/games-assets/v2/spelling-bee.4cb046f7c95f15a4188f27bdda945fb9312730b1.js:1:42219
    at https://www.nytimes.com/games-assets/v2/spelling-bee.4cb046f7c95f15a4188f27bdda945fb9312730b1.js:1:41798
    at Object.dispatch (https://www.nytimes.com/games-assets/v2/react-bundle.4cb046f7c95f15a4188f27bdda945fb9312730b1.js:288:620)
    at https://www.nytimes.com/games-assets/v2/spelling-bee.4cb046f7c95f15a4188f27bdda945fb9312730b1.js:1:95259
    at https://www.nytimes.com/games-assets/v2/spelling-bee.4cb046f7c95f15a4188f27bdda945fb9312730b1.js:1:1357
    at async Promise.all (index 1)

and this:
foundation.4cb046f7c95f15a4188f27bdda945fb9312730b1.js:1 local storage failure: DOMException: Failed to execute 'setItem' on 'Storage': Setting the value of 'sb-today' exceeded the quota.
    at E (https://www.nytimes.com/games-assets/v2/spelling-bee.4cb046f7c95f15a4188f27bdda945fb9312730b1.js:1:2505)
    at o (https://www.nytimes.com/games-assets/v2/spelling-bee.4cb046f7c95f15a4188f27bdda945fb9312730b1.js:1:984)
    at https://www.nytimes.com/games-assets/v2/spelling-bee.4cb046f7c95f15a4188f27bdda945fb9312730b1.js:1:1362
    at async Promise.all (index 1)

A batch of CORB errors, and these:
Error with Permissions-Policy header: Unrecognized feature: 'interest-cohort'.
Error with Permissions-Policy header: Unrecognized feature: 'interest-cohort'.
Error with Permissions-Policy header: Unrecognized feature: 'interest-cohort'.

**Comment by marknelson** [comment]
If you can't reproduce it, it will be hard to fix, so if so, I guess you can close this out.

**Comment by marknelson** [comment]
Oh, so I cleared all the NYT local storage in the browser and boom - Assistant is working! I don't know if this gives you a root cause but it definitely resolves it for me, I know how to fix it now.

**Comment by draber** [comment]
That explains a lot. Synchronization in Spelling Bee works in a way, that the app first makes an asynchronous call to the server to download the list of words already found and writes them to `localStorage::sb-today`. It then builds the word list etc. SBA waits for the [wordlist to be present](https://github.com/draber/draber.github.io/blob/main/src/app/modules/app.js#L83) before it eventually fetches the data from `sb-today` and builds its UI.

In other words, you did the right thing though it should be sufficient to delete `sb-today`. There isn't anything else I could do unfortunately, but I'm glad this is resolved now.
